2022-04-monthly-report.org

```diff
@@ -0,0 +1,391 @@
+#+title:  Monthly Report for April 2022
+#+author: Philip Herron
+#+date:   2022-05-02
+
+** Overview
+
+Thanks again to [[https://opensrcsec.com/][Open Source Security, inc]] and [[https://www.embecosm.com/][Embecosm]] for their ongoing support for this project.
+
+*** Milestone Progress
+
+Four months into 2022 that was quick! For gcc rust we have been working  diligently on our current milestone of imports and visibility as we have mentioned before this milestone breaks down into a few tracks:
+
+1. module support
+2. complex paths
+3. use statements
+4. metadata exports
+5.privacy checks
+
+Much of this milestone was already completed by the community and from overlapping work in previous milestones. So the remaining work is:
+
+1. metadata exports
+2. privacy checks
+3. use statements
+
+We have split the work up for the milestone between [[https://github.com/philberty][Philbert]] and [[https://github.com/CohenArthur][Arthur Cohen]] so that Arthur is working on Privacy checks which have overlapping work in use statements, so that we can handle AST::SimplePaths and resolve them. Philbert has taken advantage of the work Arthur is doing so that I could dedicate myself to bug fixing since Arthur is making good progress. This meant we got all the bugs fixed to allow us to support generating slices from arrays required for our goal test case. With more time for bug fixing, we hope to close this goal test case out soon. However, testing the compiler against actual code is very valuable to find the gaps early, and we want to continue to invest time into this as best we can.
```

Comment:
> Philbert has taken advantage of the work Arthur is doing so that I could dedicate myself to bug fixing

nit: This sounds a bit weird since you're using third person and then first person in the same sentence. I'd say stick to one or the other? You've been using third person throughout the paragraph so I guess it would make sense to keep it like that here as well, something like "Philbert has taken advantage of the work Arthur is doing to dedicate himself to bug fixing"?

Reply:
Resolved with Philbert over instant messaging since he's in the train in Mordor without an internet connection